### PR TITLE
feat: add support for delegation deposit reclaim

### PR DESCRIPTION
**Pull request by Diiaablo95**
# Add support for delegation deposit reclaim

This PR fixes https://github.com/KILTprotocol/ticket/issues/1685 by adding a new function on `DelegationNode` and the relative chain file to generate a `SubmittableExtrinsic` for the `reclaim_deposit` chain function.

This PR targets `develop` commit https://github.com/KILTprotocol/mashnet-node/commit/fc56483437001e68174f47b424fc0c6b01ad35be and `master` commit https://github.com/KILTprotocol/mashnet-node/commit/dff5ae8646753484cf594cd9af1184bd93bd405e (integration tests for this branch are failing as delegation deposits have not yet been merged).

Companion PR (already merged on `develop`) -> https://github.com/KILTprotocol/mashnet-node/pull/295

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [x] I have documented the changes (where applicable)

### CODE REVIEW

**Review by Dudleyneedham — CHANGES_REQUESTED** [review]
Some minor changes

**Comment by Dudleyneedham** [comment]
The tests are also failing.

**Comment by Diiaablo95** [comment]
The issues with tests is because we are mistakenly referring to the `latest-develop` as the latest master release, which does not have delegation deposits yet. @weichweich can we fix that, so that I can re-trigger the pipeline and see if it goes through?

File: packages/core/src/delegation/DelegationNode.ts [hunk 1]
```diff
@@ -64,7 +65,7 @@ export class DelegationNode implements IDelegationNode {
   public readonly id: IDelegationNode['id']
   public readonly hierarchyId: IDelegationNode['hierarchyId']
   public readonly parentId?: IDelegationNode['parentId']
-  public readonly childrenIds: Array<IDelegationNode['id']> = []
+  private childrenIdentifiers: Array<IDelegationNode['id']> = []
```
[the remark this answers is not in the document]
Comment (Diiaablo95) [reply packages/core/src/delegation/DelegationNode.ts:68]: Yes but I had to make `childrenIds` a public getter, while also having a private variables, and they could not be called the same. Plus, using a leading `_` would trigger lint warnings. I did not find any other way to expose what I needed to expose.

[the remark this answers is not in the document]
Comment (tjwelde) [reply packages/core/src/delegation/DelegationNode.ts:68]: Why do they need to be private? (also note, that after transpiling to javascript, it is not private anymore, just a hint for people working with typescript)

**Review by tjwelde — APPROVED** [review]
`childrenIds` could be kept public (just not readonly), otherwise LGTM

File: packages/core/src/delegation/DelegationNode.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Diiaablo95) [reply packages/core/src/delegation/DelegationNode.ts:68]: Yeah the idea is to follow, where possible, object-oriented encapsulation. Ok that with JS we lose all this nice info, but for TS developers it should not be possible to alter the array of children without revoking them. That is why I did not want to publicly expose it.

**Review by Dudleyneedham — APPROVED** [review]
LGTM

**Pull request merged**